index.ts

```diff
@@ -42,7 +42,7 @@ export const isWebPage = once((): boolean =>
 
 /** Indicates whether the code is being run in extension contexts that have access to the chrome API */
 export const isExtensionContext = once((): boolean =>
-	typeof globalThis.chrome?.extension === 'object',
+	typeof globalThis.chrome?.runtime?.id === 'string',
```

Comment:
Offscreen documents exclusively have access to the `chrome.runtime` APIs, so I had to change this detection.